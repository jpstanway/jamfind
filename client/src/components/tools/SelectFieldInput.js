import React from "react";
import classnames from "classnames";
import PropTypes from "prop-types";

const SelectFieldInput = ({
  label,
  name,
  options,
  value,
  onChange,
  disabled,
  error,
  info
}) => {
  const selectOptions = options.map(option => (
    <option key={option.label} value={option.value}>
      {option.label}
    </option>
  ));

  return (
    <div className="form-group">
      {label && <label htmlFor={name}>{label}</label>}
      <select
        name={name}
        id={name}
        value={value}
        onChange={onChange}
        className={classnames("form-control form-control-lg", {
          "is-invalid": error
        })}
      >
        {selectOptions}
      </select>
      {info && <small className="form-text text-muted">{info}</small>}
      {error && <div className="invalid-feedback">{error}</div>}
    </div>
  );
};

SelectFieldInput.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.array.isRequired,
  disabled: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.string,
  info: PropTypes.string
};

export default SelectFieldInput;
