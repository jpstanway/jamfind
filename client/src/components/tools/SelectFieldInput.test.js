import React from "react";
import { shallow } from "enzyme";
import SelectFieldInput from "./SelectFieldInput";

describe("<SelectFieldInput />", () => {
  let mockOnChangeHandler, mockOptions, mockInputData;

  beforeEach(() => {
    mockOnChangeHandler = jest.fn();
    mockOptions = [
      { label: "option 1", value: "1" },
      { label: "option 2", value: "2" },
      { label: "option 3", value: "3" },
    ];
    mockInputData = {
      label: "test",
      name: "test",
      options: mockOptions,
      value: "",
      onChange: mockOnChangeHandler,
      error: "test",
      info: "test",
    };
  });

  it("renders select input", () => {
    expect(shallow(<SelectFieldInput {...mockInputData} />)).toMatchSnapshot();
  });

  it("can change value of input", () => {
    const wrapper = shallow(<SelectFieldInput {...mockInputData} />);
    wrapper
      .find("select")
      .simulate("change", { target: { name: "test", value: "2" } });
    expect(mockOnChangeHandler.mock.calls.length).toBe(1);
  });
});
