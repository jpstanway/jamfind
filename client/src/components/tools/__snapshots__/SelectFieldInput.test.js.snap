// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectFieldInput /> renders select input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectFieldInput
    error="test"
    info="test"
    label="test"
    name="test"
    onChange={[MockFunction]}
    options={
      Array [
        Object {
          "label": "option 1",
          "value": "1",
        },
        Object {
          "label": "option 2",
          "value": "2",
        },
        Object {
          "label": "option 3",
          "value": "3",
        },
      ]
    }
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="test"
        >
          test
        </label>,
        <select
          className="form-control form-control-lg is-invalid"
          id="test"
          name="test"
          onChange={[MockFunction]}
          value=""
        >
          <option
            value="1"
          >
            option 1
          </option>
          <option
            value="2"
          >
            option 2
          </option>
          <option
            value="3"
          >
            option 3
          </option>
        </select>,
        <small
          className="form-text text-muted"
        >
          test
        </small>,
        <div
          className="invalid-feedback"
        >
          test
        </div>,
      ],
      "className": "form-group",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "htmlFor": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="1"
            >
              option 1
            </option>,
            <option
              value="2"
            >
              option 2
            </option>,
            <option
              value="3"
            >
              option 3
            </option>,
          ],
          "className": "form-control form-control-lg is-invalid",
          "id": "test",
          "name": "test",
          "onChange": [MockFunction],
          "value": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "option 1",
            "nodeType": "host",
            "props": Object {
              "children": "option 1",
              "value": "1",
            },
            "ref": null,
            "rendered": "option 1",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "option 2",
            "nodeType": "host",
            "props": Object {
              "children": "option 2",
              "value": "2",
            },
            "ref": null,
            "rendered": "option 2",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "option 3",
            "nodeType": "host",
            "props": Object {
              "children": "option 3",
              "value": "3",
            },
            "ref": null,
            "rendered": "option 3",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "form-text text-muted",
        },
        "ref": null,
        "rendered": "test",
        "type": "small",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "invalid-feedback",
        },
        "ref": null,
        "rendered": "test",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="test"
          >
            test
          </label>,
          <select
            className="form-control form-control-lg is-invalid"
            id="test"
            name="test"
            onChange={[MockFunction]}
            value=""
          >
            <option
              value="1"
            >
              option 1
            </option>
            <option
              value="2"
            >
              option 2
            </option>
            <option
              value="3"
            >
              option 3
            </option>
          </select>,
          <small
            className="form-text text-muted"
          >
            test
          </small>,
          <div
            className="invalid-feedback"
          >
            test
          </div>,
        ],
        "className": "form-group",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "htmlFor": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="1"
              >
                option 1
              </option>,
              <option
                value="2"
              >
                option 2
              </option>,
              <option
                value="3"
              >
                option 3
              </option>,
            ],
            "className": "form-control form-control-lg is-invalid",
            "id": "test",
            "name": "test",
            "onChange": [MockFunction],
            "value": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "option 1",
              "nodeType": "host",
              "props": Object {
                "children": "option 1",
                "value": "1",
              },
              "ref": null,
              "rendered": "option 1",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "option 2",
              "nodeType": "host",
              "props": Object {
                "children": "option 2",
                "value": "2",
              },
              "ref": null,
              "rendered": "option 2",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "option 3",
              "nodeType": "host",
              "props": Object {
                "children": "option 3",
                "value": "3",
              },
              "ref": null,
              "rendered": "option 3",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "form-text text-muted",
          },
          "ref": null,
          "rendered": "test",
          "type": "small",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "invalid-feedback",
          },
          "ref": null,
          "rendered": "test",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
